inst/templates/ad_advs.R

```diff
@@ -0,0 +1,271 @@
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#readdata)
+
+# Read data
+vs_metabolic <- admiralmetabolic::vs_metabolic
+adsl <- admiral::admiral_adsl
+
+# Convert SAS missing character values to NA
+advs <- vs_metabolic %>%
+  convert_blanks_to_na()
+adsl <- adsl %>%
+  convert_blanks_to_na()
+
+# Merge ADSL variables (stored in `adsl_vars`) needed for ADVS
+advs <- advs %>%
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = get_admiral_option("subject_keys")
+  )
+
+# Define parameter look-up table used for merging parameter codes to ADVS
+param_lookup <- tribble(
+  ~VSTESTCD, ~PARAMCD, ~PARAM, ~PARAMN, ~PARCAT1, ~PARCAT1N,
+  "HEIGHT", "HEIGHT", "Height (cm)", 1, "Anthropometric measurement", 1,
+  "WEIGHT", "WEIGHT", "Weight (kg)", 2, "Anthropometric measurement", 1,
+  "BMI", "BMI", "Body Mass Index(kg/m^2)", 3, "Anthropometric measurement", 1,
+  "HIPCIR", "HIPCIR", "Hip Circumference (cm)", 4, "Anthropometric measurement", 1,
+  "WSTCIR", "WSTCIR", "Waist Circumference (cm)", 5, "Anthropometric measurement", 1,
+  "DIABP", "DIABP", "Diastolic Blood Pressure (mmHg)", 6, "Vital Sign", 2,
+  "PULSE", "PULSE", "Pulse Rate (beats/min)", 7, "Vital Sign", 2,
+  "SYSBP", "SYSBP", "Systolic Blood Pressure (mmHg)", 8, "Vital Sign", 2,
+  "TEMP", "TEMP", "Temperature (C)", 9, "Vital Sign", 2
+)
+
+# Add parameter (PARAMCD) info to enable later ADVS derivations. Additional
+# parameter information will be merged again, after all AVDS derivations are
+# completed.
+
+# See function documentation for `derive_vars_merged_lookup()`:
+# (https://pharmaverse.github.io/admiral/reference/derive_vars_merged_lookup.html)
+advs <- advs %>%
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup,
+    new_vars = exprs(PARAMCD),
+    by_vars = exprs(VSTESTCD)
+  )
+
+
+# Derive Date/Time and Analysis Day ----
+# See the "Derive/Impute Numeric Date/Time and Analysis Day" vignette section
+# for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#datetime)
+
+# Add vital sign analysis date (ADT) and treatment start date (TRTSDT)
+advs <- advs %>%
+  derive_vars_dt(new_vars_prefix = "A", dtc = VSDTC)
+advs <- advs %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT))
+
+
+# Derive visit info ----
+# See the "Visit and Period Variables" vignette for more information:
+# (https://pharmaverse.github.io/admiral/articles/visits_periods.html)
+
+# Derive analysis time point (ATPT, ATPTN) and analysis visit (AVISIT, AVISITN)
+advs <- advs %>%
+  mutate(
+    ATPT = VSTPT,
+    ATPTN = VSTPTNUM,
+    AVISIT = case_when(
+      is.na(VISIT) ~ NA_character_,
+      str_detect(VISIT, "SCREEN|UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      TRUE ~ str_to_title(VISIT)
+    ),
+    AVISITN = case_when(
+      VISIT == "BASELINE" ~ 0,
+      str_detect(VISIT, "WEEK") ~ as.integer(str_extract(VISIT, "\\d+")),
+      TRUE ~ NA_integer_
+    )
```

Comment:
Needs to be revised as `AVISIT/AVISITN` are always blank for `HEIGHT` (measured at `SCREENING 1` only).

Reply:
Good catch! I don't want to deviate to much from the standard admiral BDS template, so my suggestion would be to have a added check for height. 
```
advs %>%
    mutate(
        ATPT = VSTPT,
        ATPTN = VSTPTNUM,
        AVISIT = case_when(
            is.na(VISIT) ~ NA_character_,
            VSTESTCD == "HEIGHT" & VISIT == "SCREENING 1" ~ "Screening",
            str_detect(VISIT, "SCREEN|UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
            TRUE ~ str_to_title(VISIT)
        ),
        AVISITN = case_when(
            VISIT == "BASELINE" ~ 0,
            VSTESTCD == "HEIGHT" & VISIT == "SCREENING 1" ~ -1,
            str_detect(VISIT, "WEEK") ~ as.integer(str_extract(VISIT, "\\d+")),
            TRUE ~ NA_integer_
        )
    ) 
```

Reply:
To be honest, I thought it would be "Baseline" as well, but "Screening" is fine too.

Reply:
I don't know precisely how AVISIT should be defined, but the ADAMIG states "AVISIT should be unique for a given analysis visit window.". While not entirely clear, I am interpreting this as, "The AVISIT cannot be the same for visits defined as different visit windows such as screening and baseline."

Reply:
In my opinion it is definitely CDISC compliant to have VISIT == 'SCREENING' and VISIT == 'BASELINE' on the SDTM side, but define a combined **analysis** visit window of AVISIT == 'BASELINE' on ADaM.
As you cited: "AVISIT should be unique for a given **analysis** visit window." But the **analysis** visit window can be different than the window on the collection side.

Reply:
Thanks for your input @kathrinflunkert ! Reading it again now, I agree with you! I'll go ahead and change it to baseline :)